### DISCUSSION TITLE
how to get devicetree label inside kconfig

### DISCUSSION THREAD

**Discussion by AnanthJaisankar** [zephyrproject-rtos/zephyr#26188] [General] [answered]
previously i used dt_str_val now it is changed and i unable to find any way ,please help me to do it.
Thanks

**Comment by mbolivar-nordic** [comment, marked as answer]
Hi there, there is no longer a general purpose way to do this but can you say more about your use case?

If you're looking for a label property, I assume it's because the goal is to pass it to `device_get_binding()` ultimately, and the reason to put it into Kconfig is to be able to override it. In that case, I would suggest getting the struct device from the devicetree node directly, and using devicetree overlays to override it as necessary. This is the recommended way within upstream Zephyr going forward:

https://docs.zephyrproject.org/latest/guides/dts/howtos.html

If that's not the use case, perhaps we're missing some features; any information would be useful. Thanks.
